### PR TITLE
Dispatch `getData` callbacks on FRepo callback queue

### PR DESCRIPTION
**Pull request by IanWyszynski**
https://github.com/firebase/firebase-ios-sdk/issues/8245

### CODE REVIEW

File: FirebaseDatabase/Sources/Core/FRepo.m [hunk 1]
```diff
@@ -525,11 +525,13 @@ - (void)getData:(FIRDatabaseQuery *)query
     FQuerySpec *querySpec = [query querySpec];
     id<FNode> node = [self.serverSyncTree getServerValue:[query querySpec]];
     if (node != nil) {
-        block(nil, [[FIRDataSnapshot alloc]
-                       initWithRef:query.ref
-                       indexedNode:[FIndexedNode
-                                       indexedNodeWithNode:node
-                                                     index:querySpec.index]]);
+        dispatch_async(self.config.callbackQueue, ^{
```
Comment (schmidt-sebastian) [inline FirebaseDatabase/Sources/Core/FRepo.m:528]: Can you use `eventRaiser`? https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDatabase/Sources/Core/FRepo.m#L184

File: Package.resolved [hunk 2]
```diff
@@ -0,0 +1,115 @@
+{
```
Comment (schmidt-sebastian) [inline Package.resolved:1]: Please remove

**Review by paulb777 — COMMENTED** [review]
@IanWyszynski Is there anything blocking this PR's progress? If not, please update the Changelog to release in 8.4.0 and merge.

**Comment by IanWyszynski** [comment]
> @IanWyszynski Is there anything blocking this PR's progress? If not, please update the Changelog to release in 8.4.0 and merge.

@schmidt-sebastian, could you approve this?

File: FirebaseDatabase/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# v8.4.0
+- [fixed] FirebaseDatabase `getData()` callbacks should be called on the main thread. (#8247)
```
Comment (schmidt-sebastian) [inline FirebaseDatabase/CHANGELOG.md:2]: s/should be/are now/

**Review by schmidt-sebastian — APPROVED** [review]

File: FirebaseDatabase/CHANGELOG.md (hunk 3 above)
Comment (paulb777) [reply FirebaseDatabase/CHANGELOG.md:2]: Also the 8.4.0 code freeze is past. 8.5.0 is targeted for 7/27

**Pull request merged**